packages/core/src/execution-engine/node-execution-context/utils/signature-helpers.ts

```diff
@@ -0,0 +1,34 @@
+import crypto from 'crypto';
+import type { Request } from 'express';
+
+import { WAITING_TOKEN_QUERY_PARAM } from '../../../constants';
+
+/**
+ * Generate signature token from url and secret
+ */
+export function generateSignatureToken(url: string, secret: string) {
+	const token = crypto.createHmac('sha256', secret).update(url).digest('hex');
+	return token;
+}
+
+/**
+ * Validate signature token in the request
+ */
+export function validateSignatureInRequest(req: Request, secret: string) {
+	try {
+		const token = req.query[WAITING_TOKEN_QUERY_PARAM];
```

Comment:
nit: Could be named `actualToken`. In contrast to `expectedToken`. Or even better, `actualSignature`, since it's not a token